### PR TITLE
PAYARA-3318 Restrict permitted HTTP method for FORM based auth

### PR DESCRIPTION
**Pull request by jbee**
The ticket requested configuration either based on a system property or as part of a specific configuration.

I decided to use the system property as this communicates clearly that this is a general "global" on/off switch that applies to all instances and applications. To me this is the most sensible way to implement this feature as it prevents unintentionally permitting any HTTP method for some instances or apps due to using wrong configuration scope.

If customers want to apply the system property cluster wide there is already a mechanism build into the admin GUI that allows to do so: `server => Properties => System Properties (Add Property)`. This also allows to set or change the property after startup. 

For the property name I decided for `fish.payara.permittedFormBasedAuthHttpMethods` as we (mostly) use the `fish.payara.` with no further dot levels.

### CODE REVIEW

File: appserver/web/web-core/src/main/java/org/apache/catalina/authenticator/FormAuthenticator.java [hunk 1]
```diff
@@ -148,6 +153,12 @@ public boolean authenticate(HttpRequest request, HttpResponse response, LoginCon
         
         // Is this the action request from the login page?
         boolean loginAction = requestURI.startsWith(contextPath) && requestURI.endsWith(FORM_ACTION);
+        if (loginAction) {
```
Comment (arjantijms) [inline appserver/web/web-core/src/main/java/org/apache/catalina/authenticator/FormAuthenticator.java]: It's a bit easier to see what's going on if you combine the tests:

```java
if (loginAction && !isPermittedHttpMethod(hreq.getMethod())
```

File: appserver/web/web-core/src/main/java/org/apache/catalina/authenticator/FormAuthenticator.java [hunk 2]
```diff
@@ -148,6 +153,12 @@ public boolean authenticate(HttpRequest request, HttpResponse response, LoginCon
...
+            if (!isPermittedHttpMethod(hreq.getMethod())) {
+                forwardToErrorPage(request, response, config);
```
Comment (arjantijms) [inline appserver/web/web-core/src/main/java/org/apache/catalina/authenticator/FormAuthenticator.java]: I think you maybe just want to sent a `deny` here. The error page is often tailored for wrong input (i.e. credentials did not validate).

Comment (jbee) [reply appserver/web/web-core/src/main/java/org/apache/catalina/authenticator/FormAuthenticator.java]: Changed to deny.

**Comment by jbee** [comment]
jenkins test please

**Pull request merged**